File: CSharp/Lab2_3-2_4/Lab2_3/Circle.cs
namespace Lab2_3;

public class Circle : Shape
{
    public double Radius { get; private set; }

    public Circle(double radius)
    {
        if (radius == 0) throw new ArgumentException("Коло не може бути точкою!");
        this.Radius = Math.Abs(radius);
    }

    public override double Area() => Math.PI * (Radius * Radius);

    public override double Perimeter() => 2 * Math.PI * Radius;
}

File: Cpp/Lab2_3-2_4/Lab2_3/Circle.cpp
#include <stdexcept>
#include <cmath>
#include "Circle.h"

Circle::Circle(double radius) : radius(std::abs(radius)) {
    if(radius == 0) throw std::invalid_argument("Коло не може бути точкою!");
}

double Circle::area() {
    return M_PI * (this->radius * this->radius);
}

double Circle::perimeter() {
    return 2 * M_PI * this->radius;
}

double Circle::getRadius() {
    return this->radius;
}
